core/modules/css/KvInterfaceImplMySql.cc

```diff
@@ -0,0 +1,319 @@
+    fmQuery % value;
+    fmQuery % parentKey;
+    std::string query = fmQuery.str();
+
+    // run query
+    sql::SqlErrorObject errObj;
+    LOGF(_logger, LOG_LVL_DEBUG, "create - executing query: %1%" % query);
+    if (not _conn.runQuery(query, errObj)) {
+        switch (errObj.errNo()) {
+        default:
+            throw CssError(errObj);
+            break;
+
+        case ER_DUP_ENTRY:
+            LOGF(_logger, LOG_LVL_ERROR, "create - SQL INSERT INTO failed: %1%" % query);
+            throw KeyExistsError(errObj);
+        }
+    }
+
+    transaction.commit();
+}
+
+
+void
+KvInterfaceImplMySql::set(std::string const& key, std::string const& value) {
+    if (key.size() > kMaxKeyLength) {
+        LOGF(_logger, LOG_LVL_DEBUG, "set - rejecting; key too long: %1%" % key);
+        throw CssError("key too long");
+    }
+
+    KvTransaction transaction(_conn);
+
+    boost::format fmQuery("INSERT INTO cssData (kvKey, kvVal) VALUES ('%1%', '%2%') ON DUPLICATE KEY UPDATE kvVal='%2%';");
+    fmQuery % key;
+    fmQuery % value;
+    std::string query = fmQuery.str();
+
+    // run query
+    sql::SqlErrorObject errObj;
+    LOGF(_logger, LOG_LVL_DEBUG, "set - executing query: %1%" % query);
+    if (not _conn.runQuery(query, errObj)) {
+        LOGF(_logger, LOG_LVL_ERROR, "%1% failed with err:%2%" % query % errObj.errMsg());
+        throw CssError(errObj);
+    } else if (errObj.isSet()) {
+        throw CssError(errObj);
+    }
+
+    transaction.commit();
+}
+
+
+bool
+KvInterfaceImplMySql::exists(std::string const& key) {
+    KvTransaction transaction(_conn);
+
+    std::string val;
+    sql::SqlErrorObject errObj;
+    return _getValFromServer(key, val, errObj);
+}
+
+
+std::vector<std::string>
+KvInterfaceImplMySql::getChildren(std::string const& key) {
+    KvTransaction transaction(_conn);
+
+    // per the documentation: verify the parent key exists
+    if (not exists(key)) {
+        throw NoSuchKey(key);
+    }
+
+    boost::format fmQuery("SELECT kvKey FROM cssData WHERE parentKey='%1%';");
+    fmQuery % key;
+    std::string query = fmQuery.str();
+
+    sql::SqlErrorObject errObj;
+    sql::SqlResults results;
+    LOGF(_logger, LOG_LVL_DEBUG, "getChildren - executing query: %1%" % query);
+    if (not _conn.runQuery(query, results, errObj)) {
+        LOGF(_logger, LOG_LVL_ERROR, "getChildren - %1% failed with err:%2%" % query % errObj.errMsg());
+        throw CssError((boost::format("getChildren - error:%1% from query:%2%") % errObj.errMsg() % query).str());
+    }
+
+    errObj.reset();
+    std::vector<std::string> ret;
+    if (not results.extractFirstColumn(ret, errObj)) {
+        LOGF(_logger, LOG_LVL_ERROR, "getChildren - failed to extract children from %1% failed with err:%2%" % query % errObj.errMsg());
+        throw CssError((boost::format("getChildren - failed to extract children error:%1% from query:%2%") % errObj.errMsg() % query).str());
+    }
+
+    transaction.commit();
+
+    return ret;
```

Comment:
I think this returns full path names of all children, other kvInterface implementations return relative names, we should make them all consistent.